src/app/Recordings/Filters/DurationFilter.tsx

```diff
@@ -14,61 +14,190 @@
  * limitations under the License.
  */
 
-import { Checkbox, Flex, FlexItem, TextInput } from '@patternfly/react-core';
+import { convertDurationToSeconds, DurationUnit, DurationUnitSelect } from '@app/Shared/Components/DurationUnitSelect';
+import { ActiveRecording } from '@app/Shared/Services/api.types';
+import {
+  Button,
+  ButtonVariant,
+  Checkbox,
+  Flex,
+  FlexItem,
+  HelperText,
+  HelperTextItem,
+  InputGroup,
+  InputGroupItem,
+  InputGroupText,
+  TextInput,
+  Tooltip,
+  ValidatedOptions,
+} from '@patternfly/react-core';
+import { SearchIcon } from '@patternfly/react-icons';
 import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { DURATION_INPUT_MAXLENGTH } from './const';
+
+export const CONTINUOUS_INDICATOR = 'continuous';
+
+export const filterRecordingByDuration = (recordings: ActiveRecording[], filters?: DurationRange[]) => {
+  if (!recordings || !recordings.length || !filters || !filters.length) {
```

Comment:
I think `!x || !x.length` can be collapsed down to `!x?.length`. It's a little more concise, but maybe not so readable... feel free to take or leave this style